### ISSUE TITLE
iOS simulator failing due to 'Statusgo.h' file not found

### ISSUE THREAD

**Issue by jakubgs** [status-im/status-react#11951]
@rasom and others before him have reported an issue where the simulator fails to start(or just the build fails) with:
```
/Users/joe/status-react/modules/react-native-status/ios/RCTStatus/RCTStatus.h:4:9: fatal error: 'Statusgo.h' file not found
#import "Statusgo.h"
        ^~~~~~~~~~~~
1 error generated.
```

**Comment by jakubgs** [comment]
Two remedies that we found to work is running `make nix-update-pods` or just removing `ios/Pods` folder from the repo and re-running the command, like `make run-ios`. But it's unclear why that works.

**Comment by jakubgs** [comment]
As far as I know Cocoapods are configured to symlink `react-native-status`:
https://github.com/status-im/status-react/blob/be0ad943eaf9807c79fae405c15a2a573d7ffc29/ios/Podfile#L23

Which is exactly what they do:
```
[nix-shell:~/workspace/status-react_prs_ios_PR-11949/ios/Pods/Headers/Private/react-native-status]$ ls -l
total 0
lrwxr-xr-x 1 jenkins jenkins 68 Apr  2 14:38 RCTStatus.h -> ../../../../../modules/react-native-status/ios/RCTStatus/RCTStatus.h
```

**Comment by jakubgs** [comment]
According to @rasom:
```
make clean -> make run-ios == failure
```
```
make clean -> make run-ios -> make nix-update-pods -> make run-ios == success
```
```
make clean -> make run-ios -> just removing ios/Pods -> make run-ios == success
```

**Comment by jakubgs** [comment]
He also says that "when it works i see Statusgo.framework here"

![image](https://user-images.githubusercontent.com/2212681/113422488-1aac2480-93cd-11eb-9129-ee173356a800.png)

And "when it doesn't, dir exists, but it is not shown in xcode".

![image](https://user-images.githubusercontent.com/2212681/113425351-fdc62000-93d1-11eb-9173-6798663c8989.png)

**Comment by jakubgs** [comment]
Something is different between the first `make run-ios` and the second `make run-ios` once `ios/Pods` have been updated.

But what?

**Comment by jakubgs** [comment]
I tried reproducing this issue on `macos-03` with Remmina and tmux shell but after `make clean` and `make run-ios` I got:
```
error: /Users/jenkins/repos/status-react/ios/Pods/Target Support Files/Pods-Status-StatusIm-StatusImTests/Pods-Status-StatusIm-StatusImTests.debug.xcconfig: unable to open file (in target "StatusImTests" in project "StatusIm") (in target 'StatusImTests' from project 'StatusIm')
```
Which is completely different...

**Comment by jakubgs** [comment]
Okay, I managed to reproduce it:

![image](https://user-images.githubusercontent.com/2212681/113426933-6f06d280-93d4-11eb-9351-6e80b212c7b3.png)

And `status-go` header seems correctly symlinked:
```
jenkins@macos-03.ms-eu-dublin.ci.misc:~/repos/status-react % ls -l ios/Pods/Headers/Public/react-native-status
total 0
lrwxr-xr-x  1 jenkins  jenkins  68 Apr  2 15:46 RCTStatus.h -> ../../../../../modules/react-native-status/ios/RCTStatus/RCTStatus.h
```

**Comment by jakubgs** [comment]
And the copied `Statusgo.framework` is there:
```
jenkins@macos-03.ms-eu-dublin.ci.misc:~/repos/status-react % ls -l modules/react-native-status/ios/RCTStatus 
total 72
-rw-r--r--  1 jenkins  jenkins    249 Apr  2 15:19 RCTStatus.h
-rw-r--r--  1 jenkins  jenkins  32599 Apr  2 15:19 RCTStatus.m
drwxr-xr-x@ 3 jenkins  jenkins     96 Apr  2 15:19 RCTStatus.xcodeproj
drwxr-xr-x  7 jenkins  jenkins    224 Jan  1  1970 Statusgo.framework
```
And all the files seem to be there:
```
jenkins@macos-03.ms-eu-dublin.ci.misc:~/repos/status-react % find -L modules/react-native-status/ios/RCTStatus/Statusgo.framework -type f
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Statusgo
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Resources/Info.plist
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/A/Statusgo
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/A/Resources/Info.plist
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/A/Headers/Statusgo.h
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/A/Headers/Statusgo.objc.h
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/A/Headers/ref.h
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/A/Headers/Universe.objc.h
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/A/Modules/module.modulemap
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/Current/Statusgo
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/Current/Resources/Info.plist
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/Current/Headers/Statusgo.h
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/Current/Headers/Statusgo.objc.h
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/Current/Headers/ref.h
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/Current/Headers/Universe.objc.h
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Versions/Current/Modules/module.modulemap
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Headers/Statusgo.h
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Headers/Statusgo.objc.h
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Headers/ref.h
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Headers/Universe.objc.h
modules/react-native-status/ios/RCTStatus/Statusgo.framework/Modules/module.modulemap
```

**Comment by flexsurfer** [comment]
@briansztamfater is this related https://github.com/status-im/status-react/pull/11935 ?

**Comment by briansztamfater** [comment]
@flexsurfer It could be. As this is a recent change, my thoughts are that this probably happened because of an outdated pods installation prior of updating the branch to the latest that includes #11935 and that's why ´make nix-update-pods´ fixes it. So I am wondering what happens with a fresh install of the branch, need to try.

**Comment by rasom** [comment]
@jakubgs can we build status-go before installing pods? Or we already do this?

**Comment by jakubgs** [comment]
@rasom `status-go` is built as part of the process which prepares the `ios` shell. Take a look:
https://github.com/status-im/status-react/blob/965438bc8779e473cdcd3b2609f54bb2fc704275/nix/shells.nix#L69
The shell comes from here:
https://github.com/status-im/status-react/blob/965438bc8779e473cdcd3b2609f54bb2fc704275/nix/mobile/ios/default.nix#L13-L23
And it includes `status-go-shell` via `inputsFrom`:
https://github.com/status-im/status-react/blob/965438bc8779e473cdcd3b2609f54bb2fc704275/nix/mobile/ios/default.nix#L8
Which includes `status-go` via the `shellHook` which copies it into place.
https://github.com/status-im/status-react/blob/965438bc8779e473cdcd3b2609f54bb2fc704275/nix/mobile/ios/status-go-shell.nix#L8-L10
So building it - or checking if it's already built - is already part of instantiating the Nix shell for `ios`.

**Comment by rasom** [comment]
@jakubgs so just to make it shorter, you say that `pod install` already happens after `status-go` is compiled and everything is symlinked as expected, correct?

**Comment by rasom** [comment]
@jakubgs 
just checked locally and `Copying Statusgo.framework from Nix store to ...` happens after `pod install`, that's likely the reason of the failure.

**Comment by rasom** [comment]
it can be fixed by changing order here, apparently it has reverse order of execution 
https://github.com/status-im/status-react/blob/965438bc8779e473cdcd3b2609f54bb2fc704275/nix/mobile/ios/default.nix#L21-L22

**Comment by jakubgs** [comment]
You're right, I tested it on `macos-03` and I can see it does change order:
```
jenkins@macos-03.ms-eu-dublin.ci.misc:~/repos/status-react % make shell TARGET=ios
Configuring Nix shell for target 'ios'...
Checking for modifications in ios/Pods...
No modifications detected.
Copying Statusgo.framework from Nix store to /Users/jenkins/repos/status-react/modules/react-native-status/ios/RCTStatus
No changes detected.
Done in: 2.941s

[nix-shell:~/repos/status-react]$ exit
```
```
jenkins@macos-03.ms-eu-dublin.ci.misc:~/repos/status-react % make shell TARGET=ios                                                                                
Configuring Nix shell for target 'ios'...
Copying Statusgo.framework from Nix store to /Users/jenkins/repos/status-react/modules/react-native-status/ios/RCTStatus
Checking for modifications in ios/Pods...
No modifications detected.
No changes detected.
Done in: 1.187s

[nix-shell:~/repos/status-react]$ exit
```

**Comment by jakubgs** [comment]
Nice, a manual nightly build with this fix included in #11967 worked: https://ci.status.im/job/status-react/job/manual/138/
